meta-mender-qemu/scripts/docker/extract_fs

```diff
@@ -3,23 +3,21 @@
 set -e
 
 failure() {
-    echo "Failed! Make sure you are using a privileged container." 1>&2
+    echo "Failed!" 1>&2
     cleanup
     exit 1
 }
 
-cleanup() {
-    # Ignore errors in cleanup.
-    set +e
-    kpartx -d $DEV
-    losetup -d $DEV
-}
-
 trap failure ERR
-trap cleanup EXIT
 
-DEV=$(losetup -f)
-IMG_PREFIX=${1}
+# Helpers from mender-convert
+disk_get_part_value() {
+    echo "$(partx -o ${3} -g -r --nr ${2} ${1})"
+}
+disk_extract_part() {
+    dd if=$1 of=$4 skip=$2 bs=512 count=$3 status=none
```

Comment:
Although I haven't tested this line specifically, I remember in previous experiments `bs=512` being extremely slow compared to something like `bs=1M`. Might I suggest this instead:

```
dd if=$1 of=$4 skip=${2}b bs=1M count=${3}b status=none iflag=count_bytes,skip_bytes
```

This uses GNU extensions, but I guess it should be alright here. `b` (something confusingly) means 512 multiplicand.

Reply:
Thanks! I went ahead in porting this suggestion back to `mender-convert`.